snapcraft/internal/repo/apt_cache.py

```diff
@@ -252,7 +265,10 @@ def unmark_packages(self, unmark_names: Set[str]) -> None:
         skipped_essential = set()
         skipped_filtered = set()
 
-        for package in self.cache:
+        for package in self.cache.get_changes():
```

Comment:
what is the implication of this change?

Reply:
It now iterates only over marked packages instead of all packages -- which seems correct because we're unmarking packages.